lms/djangoapps/course_api/api.py

```diff
@@ -19,6 +19,8 @@ def get_effective_user(requesting_user, target_username):
     """
     if target_username == requesting_user.username:
         return requesting_user
+    elif target_username == '' and requesting_user.username:
```

Comment:
Yeah.  I opt for eliminating code (2nd clause in this case) when it's not needed.